From 214df6ca480a21932f79631be90f875f3d28efc7 Mon Sep 17 00:00:00 2001
From: Assam Boudjelthia <assam.boudjelthia@qt.io>
Date: Wed, 15 Oct 2025 12:44:13 +0300
Subject: [PATCH 181/239] Android: skip test cases failing to acquire deadlock
 protector

It crashes when failing to acquire UI thread deadlock protector
that's acquired by input context. The reason this happens is
that some QML tests have a TextEdit or TextInput based elements
that ends up doing some operation that blocks the UI thread
even after the test case is finished and then when a next
test case tries to create a window surface it finds the UI
thread blocked. Skip these for now, otherwise a solution
needs to be found under QTBUG-139400.

This includes:
* tst_qquickwidget::focusOnClickInProxyWidget
* tst_QQuickContextMenu::textControlsMenuKey()
* tst_QQuickContextMenu::mouseAreaUnderTextArea()

Fixes: QTBUG-141161
Task-number: QTBUG-139400
Pick-to: 6.8 6.5
Change-Id: I61eed34d79aaa88c46937610dec962dcecf5c716
Reviewed-by: Rami Potinkara <rami.potinkara@qt.io>
Reviewed-by: Soheil Armin <soheil.armin@qt.io>
(cherry picked from commit e3fa4de32018f5b6c897b9df51b35d4a38c0756d)
---
 tests/auto/quickwidgets/qquickwidget/tst_qquickwidget.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/auto/quickwidgets/qquickwidget/tst_qquickwidget.cpp b/tests/auto/quickwidgets/qquickwidget/tst_qquickwidget.cpp
index a4c03ddaa1..1c4ae02664 100644
--- a/tests/auto/quickwidgets/qquickwidget/tst_qquickwidget.cpp
+++ b/tests/auto/quickwidgets/qquickwidget/tst_qquickwidget.cpp
@@ -1129,6 +1129,9 @@ void tst_qquickwidget::focusOnClick()
 #if QT_CONFIG(graphicsview)
 void tst_qquickwidget::focusOnClickInProxyWidget()
 {
+#ifdef Q_OS_ANDROID
+    QSKIP("Crashes on Android, see QTBUG-139400");
+#endif
     QGraphicsScene scene(0,0,400,400);
 
     QGraphicsView view1(&scene);
-- 
2.51.2

